test/index.spec.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const chaiString = require('chai-string')
+const expect = chai.expect
+chai.use(dirtyChai)
+chai.use(chaiString)
+
+const ipfs = require('ipfs')
+const DaemonFactory = require('ipfsd-ctl')
+const crypto = require('libp2p-crypto')
+const { fromB58String } = require('multihashes')
+
+const ipns = require('../src')
+
+const df = DaemonFactory.create({ type: 'proc', exec: ipfs })
+
+describe('ipns', function () {
+  this.timeout(20 * 1000)
+
+  const cid = 'QmWEekX7EZLUd9VXRNMRXW3LXe4F6x7mB8oPxY5XLptrBq'
+
+  let ipfs = null
+  let ipfsd = null
+  let ipfsId = null
+  let rsa = null
+
+  const spawnDaemon = (cb) => {
+    df.spawn({ initOptions: { bits: 512 } }, (err, _ipfsd) => {
+      expect(err).to.not.exist()
+      ipfsd = _ipfsd
+      ipfs = ipfsd.api
+
+      ipfs.id((err, id) => {
+        if (err) {
+          throw err
+        }
+
+        ipfsId = id
+        cb()
+      })
+    })
+  }
+
+  before(function (done) {
+    crypto.keys.generateKeyPair('RSA', 2048, (err, keypair) => {
+      expect(err).to.not.exist()
+      rsa = keypair
+
+      spawnDaemon(done)
+    })
+  })
+
+  after(function (done) {
+    ipfsd.stop(() => done())
```

Comment:
If spawn fails you'll also get an error here because `ipfsd` will still be null. It's not a big problem just more noise in your console.